### PR TITLE
[TTAHUB-2367] Fix FEI Root Cause local storage bug

### PR DESCRIPTION
**Pull request by AdamAdHocTeam**
## Description of change

A bug was found during the demo of TTAHUB-23767. When updating FEI root cause on an activity report, we would properly 
propagate the root cause to all other reports and goals. However because the AR '**updatedAt**' was NOT changed. The local storage thought it had a more recent copy of the report and overwrote it.

## How to test

- Review the code.

**AR Test:**
1. Create two AR's using the same FEI goal.
2. Update the FEI on one of the AR's and save.
3. Verify both AR's have the set FEI.
4. Verify the Goal in RTR has the set FEI from the AR.
5. Check both AR's in the db to make sure they have the same **updatedAt** date.

**RTR Test:**
1. Create two AR's using the same FEI goal.
2. Update the FEI from the RTR.
3. Verify both AR's have the RTR set FEI.
4. Verify the RTR FEI value stuck on the goal.
5. Check both AR's in the db to make sure they have the same **updatedAt** date.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-2367

### Every PR

- [x] Meets issue criteria
- [x] JIRA ticket status updated
- [x] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [x] Update JIRA ticket status

### CODE REVIEW

**Comment by thewatermethod** [comment]
test failure indicates a syntax error, looks related

In the PR description, 2 places:
_5. Check both AR's in the db to make sure they have the same **createdAt** date._
should be
_5. Check both AR's in the db to make sure they have the same **updatedAt** date._
right?

**Comment by AdamAdHocTeam** [comment]
> test failure indicates a syntax error, looks related
> 
> In the PR description, 2 places: _5. Check both AR's in the db to make sure they have the same **createdAt** date._ should be _5. Check both AR's in the db to make sure they have the same **updatedAt** date._ right?

Fixed.

File: src/models/hooks/goalFieldResponse.js
Comment (thewatermethod) [inline src/models/hooks/goalFieldResponse.js]: I think we do need to check the length of "activityReportIds," it looks like at least for test purposes. (Bright side: the path is already tested)

Screenshot 2023-12-19 at 4 42 17 PM

**Review by thewatermethod — APPROVED** [review]

**Comment by thewatermethod** [comment]
approved, let's wait for it to pass CI before merging!

File: src/models/hooks/goalFieldResponse.js
Comment (AdamAdHocTeam) [reply src/models/hooks/goalFieldResponse.js]: Yea I already added that funny that I brought that up when we looked at it lol

File: src/models/hooks/goalFieldResponse.js
Comment (GarrettEHill) [inline src/models/hooks/goalFieldResponse.js:77]: @AdamAdHocTeam 

Why was this done this way and not using the model structure?
This is a backend sequelize hook, how does this have anything to do with local storage on the users browser?

Comment (GarrettEHill) [reply src/models/hooks/goalFieldResponse.js:77]: I think i understand why we need to update the date now, but not why it was done with raw SQL. using raw SQL opens us up for more parsing issue. Dates most specifically.

Comment (AdamAdHocTeam) [reply src/models/hooks/goalFieldResponse.js:77]: We cannot update the updatedAt column through the model process.

Comment (GarrettEHill) [reply src/models/hooks/goalFieldResponse.js:77]: @AdamAdHocTeam 
`await ActivityReport.update({ updatedAt }, { where: { id: activityReportIds }, silent: true });`

**Pull request merged**
